File: ios/ScaScanKit/Sources/ScaScanKit/Domain/UseCases/ScanBarcodeUseCase.swift
/// Mirrors Android's `domain.usecase.ScanBarcodeUseCase`.
public struct ScanBarcodeUseCase: Sendable {
    private let repository: NutritionRepository

    public init(repository: NutritionRepository) {
        self.repository = repository
    }

    public func callAsFunction(_ barcode: String) async throws -> NutritionFacts {
        try await repository.analyzeBarcode(barcode)
    }
}

File: android/app/src/main/java/com/mtreconsulting/scascan/domain/usecase/ScanBarcodeUseCase.kt
package com.mtreconsulting.scascan.domain.usecase

import com.mtreconsulting.scascan.data.model.NutritionFacts
import com.mtreconsulting.scascan.data.repository.NutritionRepository
import javax.inject.Inject

class ScanBarcodeUseCase @Inject constructor(
    private val repository: NutritionRepository
) {
    suspend operator fun invoke(barcode: String): Result<NutritionFacts> =
        repository.analyzeBarcode(barcode)
}
